common/add_vsphere_hosts_in_inventory.yml

```diff
@@ -21,6 +21,52 @@
     - vcenter_ssh_password is defined
     - vcenter_ssh_password
 
+- name: "Get datastores on ESXi server"
+  include_tasks: esxi_get_datastores.yml
+
+- name: "Check ESXi server has accessible datastores"
+  ansible.builtin.assert:
+    that:
+      - esxi_datastores | length > 0
+    fail_msg: "There is no accessible datastore on ESXi server"
+
+- name: "Check datastore and set temporary directory on ESXi server"
+  when:
+    - datastore is defined
+    - datastore
+  block:
+    - name: "Check '{{ datastore }}' exists on ESXi server"
+      ansible.builtin.assert:
+        that:
+          - esxi_datastores | length > 0
+          - esxi_datastores | selectattr('name', 'equalto', datastore) | length == 1
+        fail_msg: "Datastore '{{ datastore }}' doesn't exist on ESXi server"
```

Comment:
"Datastore '{{ datastore }}' is not in the accessible datastore list on ESXi server"